### ISSUE TITLE
Tree Select Control: Parent should be indeterminate when child is selected but parent is not

### ISSUE THREAD

**Issue by joshuatf** [woothemes/woocommerce#38232]
When selecting a child underneath a parent, we should show the indeterminate status in the parent's checkbox if the parent is not also selected.

Note that the design here is still a WIP and we should confirm with @jarekmorawski before starting on this issue.

### Screenshots

![Image](https://user-images.githubusercontent.com/10561050/237541133-299289a5-f0b1-44d2-8b27-144d4c91445f.png)

### Acceptance criteria

* [ ] When a child item is selected but the parent is not selected, and there are other child items not selected, the indeterminant status should be shown for the checkbox
* [ ] If **all** child items under a parent are selected, then the parent's checkbox should be automatically displayed as checked as well. 
* [ ] When the parent is selected directly, the parent's checkbox should be checked regardless of child selection
* [ ] When neither the parent and children have not been selected, an unchecked checkbox should be shown

**Comment by pmcpinto** [comment]
> Note that the design here is still a WIP and we should confirm with @jarekmorawski before starting on this issue.

Do we still need design input on this issue?

**Comment by louwie17** [comment]
> Do we still need design input on this issue?

Yes, @j111q are you able to confirm the above? as there is no example in Figma that I saw.

**Comment by j111q** [comment]
Reviewed and edited the AC above directly to add/clarify these two points:

>  When a child item is selected but the parent is not selected, and there are other child items not selected, the indeterminant status should be shown for the checkbox
>  If all child items under a parent are selected, then the parent's checkbox should be automatically displayed as checked as well.

Regarding the screenshot above:

<img width="863" alt="image" src="https://github.com/woocommerce/woocommerce/assets/65582817/d6deb24b-4391-4711-a226-6dd86a35e3bc">

It should be checked, not indeterminate, since everything within that parent category is already checked. Does that makes sense?

**Comment by louwie17** [comment]
> It should be checked, not indeterminate, since everything within that parent category is already checked. Does that makes sense?

@j111q this does make sense, but the reason for the previous criteria is because the parent item is a category as well. So in this case do we auto add the parent category if all the children are selected?
We previously decided against this, as folks may want to just select the children as categories.
In this case do we want to keep the indeterminate state for the parent: if the parent is not selected but all the children are?

**Comment by j111q** [comment]
Hmm I see what you’re saying. 🤔

I’m trying to think about this through a real use-case, eg. let’s say you’re creating a Product Ad and you’re choosing where it should be shown. The dropdown shows Continents > Regions > Countries. If I select “Asia”, I would expect all the countries in Asia to appear in the pills above, but there should _not_ be a pill showing first-parent name “Asia”, or the second-parent names eg. “Central Asia” / “East Asia”. And I think I would expect both parent categories to be checked in the dropdown.  

So TLDR I think, _yes_, we should auto-check the parent category if all child categories are selected, even if the user didn’t select the parent category directly. 

Is there a different use case I’m missing where this wouldn’t make sense though? Tricky!

**Comment by pmcpinto** [comment]
I'm going to close this one. It refers to the new editor that will be removed.
